mods/weather/init.lua

```diff
@@ -114,22 +103,93 @@ local function update_clouds()
 			speed = {x = n_speedx * 4, z = n_speedz * 4},
 		})
 		-- now adjust the shadow intensity
-		player:set_lighting({ shadows = { intensity = 0.7 * (1 - density) } })
+		player:set_lighting({
+			shadows = { intensity = 0.7 * (1 - density) }
+		})
 	end
 end
 
+local function purge_effects(player)
+	-- reset potentially touched values to their defaults
+	if randomize_clouds then
+		player:set_clouds({
+			density = 0.4,
+			thickness = 16,
+			speed = { x = 0, z = -2 }
+		})
+	end
+	player:set_lighting({
+		shadows = { intensity = 0 }
+	})
+end
 
-local function cyclic_update()
-	update_clouds()
-	minetest.after(CYCLE, cyclic_update)
+-- Define API hooks
+
+-- override to set state for newly joined players
+weather.enable_on_join = true
+
+-- returns bool for whether weather is active
+-- returns nil if player is offline or weather is disabled
+weather.get_enabled = function(player)
+	return playerlist[player]
```

Comment:
This API would change to returning `nil` (rather than `false`) if weather is disabled. It could be fixed using an "offline" check though.